tooling/minimize/src/ty.rs

```diff
@@ -11,26 +11,59 @@ impl<'tcx> Ctxt<'tcx> {
             let size = translate_size(layout.size());
             let align = translate_align(layout.align().abi);
             let layout = LayoutStrategy::Sized(size, align);
-            return PointeeInfo { layout, inhabited, freeze, unpin };
+
+            // Because we compute `nonfreeze_bytes` by iterating through the fields of
+            // the type in declaration, not in memory order, the order of the ranges are
+            // not necessarily sorted in ascending order.
+            let mut nonfreeze_bytes = self.nonfreeze_bytes_in_sized_ty(ty, span);
+            nonfreeze_bytes.sort_by(|a, b| a.0.cmp(&b.0));
+            let nonfreeze_bytes = nonfreeze_bytes.into_iter().collect::<List<(Offset, Offset)>>();
```

Comment:
You called the permission `Cell`, so it'd only make sense to call these `cell_bytes`.

Reply:
But the `Cell` permission is only used for shared references to interior mutable data.  Here we don't care about whether the reference is shared or mutable.

Reply:
It might be worth to plan ahead a bit here. In future, that method might also tell you which bytes are `MabyeAliased` so a name that does not fixate too much on the fact that it's originally about `UnsafeCell` would be helpful. Also note that this method is used to figure out what is `Reserved` vs `ReservedIM`, which does not involve the `Cell` permission.

Reply:
> this method is used to figure out what is Reserved vs ReservedIM, which does not involve the Cell permission.

It's about the presence of `(Unsafe)Cell`, though.

> It might be worth to plan ahead a bit here. In future, that method might also tell you which bytes are MabyeAliased so a name that does not fixate too much on the fact that it's originally about UnsafeCell would be helpful.

By `MaybeAliased` do you mean [`UnsafePinned`](https://github.com/rust-lang/rust/issues/125735) or something else?